### PR TITLE
[ Layer ] Draft of Conv2D Layer

### PR DESCRIPTION
**Pull request by jijoongmoon**
This is the first draft of 2d convolution layer

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/include/conv2d_layer.h [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  /**
+   * @brief     Save Weight & Bias Data to file
+   * @param[in] file output stream file
+   */
+  void save(std::ofstream &file);
+
+  /**
+   * @brief     forward propagation with input
+   * @param[in] in Input Tensor from upper layer
+   * @retval    Activation(W x input + B)
```
Comment (dongju-chae) [inline nntrainer/include/conv2d_layer.h:69]: Missing `status`

File: nntrainer/include/conv2d_layer.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+   * @brief     Property Enumeration
+   *            0. input shape : string
+   *            1. bias zero : bool
+   *            4. activation : bool
+   *            6. weight_decay : string (type)
+   *            7. weight_decay_lambda : float
+   *            9. filter_size : int
+   *            10. kernel_size : ( n , m )
+   *            11. stride : ( n, m )
+   *            12, padding : valid | same
```
Comment (zhoonit) [inline nntrainer/include/conv2d_layer.h:106]: missing `weight_ini` and number after 9 needs correction.

File: nntrainer/include/conv2d_layer.h (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/include/conv2d_layer.h:69]: fixed. Thanks.

File: nntrainer/include/conv2d_layer.h (hunk 2 above)
Comment (jijoongmoon) [reply nntrainer/include/conv2d_layer.h:106]: fixed. thanks.

**Pull request merged**